core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1066,50 +1066,53 @@ public static boolean isRemoteSegmentWithinLeaderEpochs(RemoteLogSegmentMetadata
         // Check for out of bound epochs between segment epochs and current leader epochs.
         Integer segmentFirstEpoch = segmentLeaderEpochs.firstKey();
         Integer segmentLastEpoch = segmentLeaderEpochs.lastKey();
+
+        // FIXME: We have to remove the below check too for `DELETE_RECORDS` API to work properly.
```

Comment:
Not clear on the usage of `isRemoteSegmentWithinLeaderEpochs` validation in the segment deletion path. Commented out the code for now, will go through the code.